filter/url_param/filterurlparam.go

```diff
@@ -0,0 +1,143 @@
+package filterurlparam
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/tsaikd/KDGoLib/errutil"
+	"github.com/tsaikd/gogstash/config"
+	"github.com/tsaikd/gogstash/config/goglog"
+	"github.com/tsaikd/gogstash/config/logevent"
+)
+
+// ModuleName is the name used in config file
+const ModuleName = "url_param"
+
+// errors
+var (
+	ErrParamParse = errutil.NewFactory("param `%q` parse error")
+)
+
+// FilterConfig holds the configuration json fields and internal objects
+type FilterConfig struct {
+	config.FilterConfig
+
+	// The field containing the user agent string.
+	Source string `json:"source"`
+
+	// Include param keys, "all_fields" or "*" include all fields
+	IncludeKeys []string `json:"include_keys"`
+	includeAll  bool     `json:"-"`
+
+	// url_decode params, "all_fields" or "*" decode all params values
+	UrlDecode []string `json:"url_decode"`
+	decodeAll bool     `json:"-"`
+
+	// prefix for param name, default: request_url_args_
+	Prefix string `json:"prefix"`
+
+	RemoveEmptyValues bool `json:"remove_empty_values"`
+}
+
+// DefaultFilterConfig returns an FilterConfig struct with default values
+func DefaultFilterConfig() FilterConfig {
+	return FilterConfig{
+		FilterConfig: config.FilterConfig{
+			CommonConfig: config.CommonConfig{
+				Type: ModuleName,
+			},
+		},
+		Source:            "request_url",
+		IncludeKeys:       []string{"*"},
+		UrlDecode:         []string{"*"},
+		Prefix:            "request_url_args_",
+		RemoveEmptyValues: true,
+	}
+}
+
+// InitHandler initialize the filter plugin
+func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeFilterConfig, error) {
+	conf := DefaultFilterConfig()
+	err := config.ReflectConfig(raw, &conf)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(conf.IncludeKeys) == 0 {
+		goglog.Logger.Fatalf("include_keys is empty configure.")
+	}
+
+	if conf.IncludeKeys[0] == "*" {
+		conf.includeAll = true
+	} else {
+		conf.includeAll = false
+	}
+
+	if len(conf.UrlDecode) > 0 && conf.UrlDecode[0] == "*" {
+		conf.decodeAll = true
+	} else {
+		conf.decodeAll = false
+	}
+
+	if conf.Prefix != "" {
+		if strings.Index(conf.Prefix, ".") >= 0 {
+			return nil, fmt.Errorf("prefix can not includ dot(\".\")")
+		}
+	}
+
+	return &conf, nil
+}
+
+// Event the main filter event
+func (f *FilterConfig) Event(ctx context.Context, event logevent.LogEvent) logevent.LogEvent {
+	message := event.GetString(f.Source)
+	var params url.Values
+	var err error
+	if up := strings.Split(message, "?"); len(up) == 1 {
```

Comment:
why not [url.Parse()](https://golang.org/pkg/net/url/#Parse)?

Reply:
I have modified it